agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/AIAgent.kt

```diff
@@ -133,6 +133,8 @@ public open class AIAgent<Input, Output>(
 
     private val pipeline = AIAgentPipeline()
 
+    override val id: String by lazy { id ?: Uuid.random().toString() }
```

Comment:
@ygyg70, thank you for the catch. Looks good! Could you please just add a comment here, to make it clear why is it important to have id calculation at runtime.